sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -287,14 +287,14 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
           new Path(metastoreRelation.catalogTable.storage.locationUri.get),
           partitionSpec)
 
+        val schema =
+          defaultSource.inferSchema(sparkSession, options, fileCatalog.allFiles())
         val inferredSchema = if (fileType.equals("parquet")) {
```

Comment:
I just noticed the boolean expression should be `fileType == "parquet"` to make it idiomatic Scala. Can you make that change?